internal/registry/registry.go

```diff
@@ -881,7 +884,7 @@ func (r *Registry) pollProviders(poll polling, pollOverrides map[peer.ID]polling
 				}
 				// Tell the ingester to remove data for the provider.
 				info.deleted = true
-			} else if sincePollingStarted >= 2*poll.retryAfter {
+			} else if sincePollingStarted >= poll.deactivateAfter {
```

Comment:
Update the comment below, because now it is not after at least one poll attempt, it is after `deactivateAfter`